float32 posx
float32 posy
---
uint8 mapx
uint8 mapy